src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -77,6 +77,15 @@ public List<DarCollection> getCollectionsForUserByRoleName(User user, String rol
           collections.addAll(getAllCollections());
           break;
         case CHAIRPERSON:
+          List<Dataset> userDatasets = datasetDAO.findDatasetsByAuthUserEmail(user.getEmail());
+          List<DarCollection> userCollections = getCollectionsByUserDacs(user);
+          userCollections.forEach(c -> {
+            Set<Dataset> relevantDatasets = c.getDatasets();
+            relevantDatasets.retainAll(userDatasets);
+            c.setDatasets(relevantDatasets);
+          });
+          collections.addAll(userCollections);
+          break;
         case MEMBER:
```

Comment:
We would want these changes to be present for both chairperson and member roles, so I would put `case MEMBER:` directly below `case CHAIRPERSON:` so both cases in the switch statement use the new block of code